ckan/plugins/interfaces.py

```diff
@@ -1717,6 +1731,20 @@ def get_resource_uploader(self):
         :param id: resource id
         :type id: string
 
+        ``delete(id)``
+
+        Perform a delete.
+
+        :param id: resource id, used to delete file
+        :param filename: can be None.
```

Comment:
A more comprehensive description would be nice, eg "The filename to use when storing a resource, may be None depending on the storage provider."